### PR TITLE
Added configurable chunkSize to HDFOutputer

### PR DESCRIPTION
**Pull request by Dr15Jones**
Also fixed the last batch not being written bug.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
@sabasehrish @knoepfel please have a look

**Comment by Dr15Jones** [comment]
@QiaoK this PR includes a fix for the end of job batch problem you had uncovered.

File: HDFOutputer.cc [hunk 1]
```diff
@@ -81,6 +82,16 @@ void HDFOutputer::outputAsync(unsigned int iLaneIndex, EventIdentifier const& iE
 void HDFOutputer::printSummary() const  {
   std::cout <<"HDFOutputer\n  total serial time at end event: "<<serialTime_.count()<<"us\n"
     "  total parallel time at end event: "<<parallelTime_.load()<<"us\n";
+
+  auto start = std::chrono::high_resolution_clock::now();
+  if (batch_ != 0) {
+    //flush the remaining data to the file
+    const_cast<HDFOutputer*>(this)->writeBatch();
```
Comment (Dr15Jones) [inline HDFOutputer.cc:89]: This fixes the reported bug.

**Comment by knoepfel** [comment]
Looks fine to me, @Dr15Jones.  I defer to any comments Saba may have.

**Comment by sabasehrish** [comment]
Looks okay to me, I am just thinking that we know 128 is not a good chunk size, should we change it to something better performing? Otherwise no more suggestions.

**Comment by Dr15Jones** [comment]
> I am just thinking that we know 128 is not a good chunk size, should we change it to something better performing?

We can. I was trying not to break old results by changing the default.

**Pull request merged**